ibm_cloud_sdk_core/authenticators/iam_assume_authenticator.py

```diff
@@ -106,7 +106,7 @@ def __init__(
     def __getattribute__(self, name: str) -> Any:
         disallowed_attrs = ['set_scope', 'set_client_id_and_secret']
         if name in disallowed_attrs:
-            raise AttributeError(f"'IAMAssumeAuthenticator' has no attribute '{name}'")
+            raise AttributeError(f"'{self.__class__.__name__}' has no attribute '{name}'")
```

Comment:
Now we won't have to care about changing this if we decide to rename the class. :)